bin/delete-empty-mirrors.js

```diff
@@ -0,0 +1,157 @@
+#!/usr/bin/env node
+
+'use strict';
+
+const program = require('commander');
+const Storage = require('storj-service-storage-models');
+const mongoose = require('mongoose');
+const { eachLimit } = require('async');
+const Config = require('../lib/config');
+
+program.version('0.0.1')
+  .option('-c, --config <path_to_config_file>', 'path to the config file')
+  .option('-u, --mongourl <mongo connection string>', 'mongo url connection string')
+  .option('-m, --lastMirror <last deleted mirror>', 'last deleted mirror from where to continue')
+  .parse(process.argv);
+
+if (!process.env.NODE_ENV) {
+  throw new Error('NODE_ENV is not set');
+}
+
+let mongourl;
+let deleteCount = 0;
+let idMirrorBeingChecked;
+let lastMirror;
+
+if (program.mongourl) {
+  mongourl = program.mongourl;
+}
+
+if (program.lastMirror) {
+  lastMirror = program.lastMirror;
+}
+
+const config = new Config(process.env.NODE_ENV, program.config);
+const storage = new Storage(
+  mongourl || config.storage.mongoUrl,
+  config.storage.mongoOpts
+);
+
+const { Shard, Mirror } = storage.models;
+
+const logStatus = () => {
+  console.log('Deleted mirrors: ', deleteCount);
+  if (idMirrorBeingChecked) {
+    console.log('Last mirror checked: ', idMirrorBeingChecked);
+  }
+};
+
+const loggerInterval = setInterval(logStatus, 4000);
+
+const deleteEmptyMirror = (mirror, onDelete, cb) => {
+  const { shardHash } = mirror;
+  Shard.findOne({ hash: shardHash }, (err, shard) => {
+    if (err) {
+      return cb(err);
+    }
+
+    if (!shard) {
+      mirror.remove(err => {
+        if (err) {
+          return cb(err);
+        } else {
+          onDelete(cb);
```

Comment:
Not sure if its a good idea to pass the cb to the onDelete function here, or to do it in the next line.

I opted for passing it just in case we had an async onDelete somewhere else and we only wanted one flow for the cb.

@sg-gs

Reply:
In this case, is not a bad option as you need the react to each iteration without coupling `deleteEmptyMirror`